### PR TITLE
ft(updatepasssword): user can update their password [#187364862]

### PR DESCRIPTION
**Pull request by Oliviier-dev**
## ℹ️ Description

As a user

- I should be able to update my password
- When the user provides valid password as well as new password his/her password will be updated

## 🧪 How can this be tested?

This should be tested on the user settings page, under the route of user/usersettings.

## 📸 Screenshots

Screenshot 2024-06-29 at 19 44 26

## 📍 Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## ✅ Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have created a loom video for the new feature or enhancement
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by tuyishimehope — DISMISSED** [review]
LGTM

**Review by shebz2023 — DISMISSED** [review]
LGTM

File: src/components/rootcomponents/Sidebar.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    icon: <MdOutlineNotificationsActive className="text-2xl" />,
+    link: '/notifications',
+  },
+  {
+    id: 5,
+    name: 'Wishlist',
+    icon: <FaRegHeart className="text-2xl" />,
+    link: '/wishlist',
+  },
+];
```
Comment (SergeInGithub) [inline src/components/rootcomponents/Sidebar.tsx]: Put this in the data folder under source. create a file there with relevant name, export it and use it here

**Review by SergeInGithub — CHANGES_REQUESTED** [review]
Nice work so far. Some reviews below @Oliviier-dev

**Review by SergeInGithub — APPROVED** [review]
LGTM

**Pull request merged**